### PR TITLE
fix: Allow on Submit for Material Request Item Required Date

### PR DESCRIPTION
**Pull request by bhattdevarsh**
### ISSUE
User not able to change/update the Required Date after Submitting, but the main(parent) document has "Allow on Submit" for Required date.

### SOLUTION
Allow user to Update the Required Date Item wise after Submitting the document.

### CODE REVIEW

File: erpnext/stock/doctype/material_request_item/material_request_item.json [hunk 1]
```diff
@@ -177,6 +177,7 @@
    "fieldtype": "Column Break"
   },
   {
+   "allow_on_submit": 1,
```
Comment (ankush) [inline erpnext/stock/doctype/material_request_item/material_request_item.json:180]: If we want to allow this then we also need to validate by adding same validation in `before_update_after_submit` methods on material request.


ref:
https://github.com/frappe/erpnext/blob/79a51b24812725a541c3accf68fdde004582da5d/erpnext/stock/doctype/material_request/material_request.py#L59-L62

https://github.com/frappe/frappe/blob/a886d793f29fb3c291620958286b52b3665c1ebc/frappe/model/document.py#L955-L961

**Review by ankush — CHANGES_REQUESTED** [review]

File: erpnext/stock/doctype/material_request_item/material_request_item.json (hunk 1 above)
Comment (bhattdevarsh) [reply erpnext/stock/doctype/material_request_item/material_request_item.json:180]: I think Validate does not run on Update, can you please explain the reason of updating Validate code? @ankush

Comment (ankush) [reply erpnext/stock/doctype/material_request_item/material_request_item.json:180]: It **should** run on update, right now it doesn't :) that's what I am asking for adding.

**Review by ankush — APPROVED** [review]

File: erpnext/stock/doctype/material_request/material_request.py [hunk 2]
```diff
@@ -83,6 +83,9 @@ def validate(self):
 		self.reset_default_field_value("set_warehouse", "items", "warehouse")
 		self.reset_default_field_value("set_from_warehouse", "items", "from_warehouse")
 
+	def before_update_after_submit(self):
+		self.validate_schedule_date()
+
```
Comment (ankush) [inline erpnext/stock/doctype/material_request/material_request.py]:
```suggestion

```

whitepsace

**Pull request merged**